src/main/java/frc/robot/commands/ArmCommand.java

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.button.CommandXboxController;
+import frc.robot.Constants.ArmConstants;
+import frc.robot.subsystems.ArmSubsystem;
+import frc.robot.subsystems.Intake;
+
+public class ArmCommand extends CommandBase {
+
+  private ArmSubsystem m_arm;
+  private CommandXboxController m_controller;
+  private Intake m_intake;
+  private Color m_color;
+
+  /** Creates a new ArmCommand. */
+  public ArmCommand(ArmSubsystem subsystem, CommandXboxController controller, Intake intake) {
+    m_arm = subsystem;
+    m_controller = controller;
+    m_intake = intake;
+    addRequirements(m_arm);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {}
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    if (m_controller.povUp().getAsBoolean()) {
+      m_arm.rotateArm(0.3);
+    } else if (m_controller.povDown().getAsBoolean()) {
+      m_arm.rotateArm(-0.3);
+    } else {
+      m_arm.rotateArm(0);
+    }
+
+    if (m_controller.x().getAsBoolean()) {
+      m_color = m_intake.getColor();
+      if (m_color == ArmConstants.kConeColor || m_color == ArmConstants.kCubeColor) {
+        m_arm.grabArm(0.01);
+      }
+    }
+
+    if (m_controller.b().getAsBoolean()){
+      m_arm.grabArm(-0.01);
+    }
+  }
```

Comment:
Why use the same command for several different button bindings? Most can be converted to separate commands that are bound to say `btnX.onPress()` by `RobotContainer`.

~~Some of this functionality can even be converted to lambda commands~~